### ISSUE TITLE
getProductPriceRange(productId) does not return range

### ISSUE THREAD

**Issue by Valekst3** [reactioncommerce/reaction#1597]
### Expected behavior
Function `getProductPriceRange` - should return range of the Product
### Actual Behavior
`Range is not returned -> range == undefined even if we have correct ProductId`
### Steps to Reproduce the Behavior
In function 

```js
Template.productSettingsGridItem.helpers({
   displayPrice: function () {
      if (this._id) {
         return ReactionProduct.getProductPriceRange(this._id).range;
      }
      return null;
   },
});
```

we recieve error in `ReactionProduct.getProductPriceRange(this._id).range`  

`getProductPriceRange(productId)` is defined in `lib/api/products.js`, `getProductPriceRange` recieves correct id but doesnt returns range, that means that there is some error in core `lib/api/product.js` file

**Comment by dshitik** [comment]
Yep. I have same problem. The database is not written changed value of the price. When i crate new product is not written price into the database for the simple product and the price range for type String for the variative product.

**Comment by Redmega** [comment]
This occurs when I have no option variants, but only an option. Price should be $10 but it is not showing if I'm not logged in to admin.

**Comment by mikemurray** [comment]
The thing is, if there are no option variants, then the returned value for `ReactionProduct.getProductPriceRange(this._id)` is not an object; so doing `ReactionProduct.getProductPriceRange(this._id).range` will result in an error.

This helper could be updated to better handle accessing a price range from `ReactionProduct.getProductPriceRange`.

**Comment by dshitik** [comment]
There is a basic product with options as well as price range. Good. But when I change the price of the product with the type:Variant, database field change only with the type of product:Variant, and the field price.range, price.min, price.max - does not change. And when i create new simple product with variants, all of these fields are not created, and this not good.
I understand the price range must be formed on the basis of prices in variants products. But its not work.
So, we have - same function not generate price range(type: String) for simple products.

**Comment by zenweasel** [comment]
I think there is more than one issue here. I feel like this is a duplicate of https://github.com/reactioncommerce/reaction/issues/1519
If I add a product with only a top level variant I can do this:

```
> import { ReactionProduct } from "/lib/api";
> ReactionProduct.getProductPriceRange("a2Dm3ThwroAmQFNQx")
0.22
```

**Comment by luisfavila** [comment]
I took @mikemurray answer and changed the code.
@zenweasel This seems to be solved by changing
```
Template.productSettingsGridItem.helpers({
   displayPrice: function () {
      if (this._id) {
         return ReactionProduct.getProductPriceRange(this._id).range;
      }
    return null;
  },
```
to
```
Template.productSettingsGridItem.helpers({
  displayPrice: function () {
    if (this._id) {
      if(typeof ReactionProduct.getProductPriceRange(this._id) === 'object')
	  	return ReactionProduct.getProductPriceRange(this._id).range;
	  else
	    ReactionProduct.getProductPriceRange(this._id);
    }
    return null;
  },
```

**Comment by kieckhafer** [comment]
This is fixed in #1633
